### PR TITLE
Feat:Add default version selection for composition expanders.

### PR DESCRIPTION
**Pull request by xiaoweim**
### Tests you have done

- [x] Run `make ready-pr` to ensure this PR is ready for review.
- [x] Perform necessary E2E testing for changed resources.

### CODE REVIEW

File: experiments/compositions/composition/tests/data/TestSimpleCompositionAddFacadeField/input.yaml [hunk 1]
```diff
@@ -108,7 +108,6 @@ spec:
   inputAPIGroup: pconfigs.facade.foocorp.com
   expanders:
   - type: jinja2
-    version: v0.0.1.alpha
```
Comment (barney-s) [inline experiments/compositions/composition/tests/data/TestSimpleCompositionAddFacadeField/input.yaml:111]: so the version is optional for now. If not specified it picks the latest ?

File: experiments/compositions/composition/internal/controller/expander_reconciler.go [hunk 2]
```diff
@@ -308,3 +313,40 @@ func (r *ExpanderReconciler) SetupWithManager(mgr ctrl.Manager, cr *unstructured
 		For(cr).
 		Complete(r)
 }
+
+func (r *ExpanderReconciler) getExpanderVersion(ctx context.Context, inputExpanderVersion string, expanderType string) (string, string) {
+	logger := log.FromContext(ctx)
+
+	var expanderVersionCR compositionv1alpha1.ExpanderVersion
+	var expanderVersion string
+	err := r.Client.Get(ctx, types.NamespacedName{Name: expanderType, Namespace: "composition-system"}, &expanderVersionCR)
```
Comment (barney-s) [inline experiments/compositions/composition/internal/controller/expander_reconciler.go:322]: please return the err here and handle it in the call site.

File: experiments/compositions/composition/internal/controller/expander_reconciler.go [hunk 3]
```diff
@@ -308,3 +313,40 @@ func (r *ExpanderReconciler) SetupWithManager(mgr ctrl.Manager, cr *unstructured
...
+	if err != nil {
+		reason := "FailedGettingExpanderVersionCRD"
+		if apierrors.IsNotFound(err) {
+			reason = "MissingFacadeCRD"
```
Comment (barney-s) [inline experiments/compositions/composition/internal/controller/expander_reconciler.go:326]: MissingExpanderVersionCR

File: experiments/compositions/composition/internal/controller/expander_reconciler.go [hunk 4]
```diff
@@ -178,9 +181,11 @@ func (r *ExpanderReconciler) Reconcile(ctx context.Context, req ctrl.Request) (c
 		// ------------------- EXPANSION SECTION -----------------------
 		logger = loggerCR.WithName(expander.Name).WithName("Expand")
 
+		expanderVersion, expanderRegistry := r.getExpanderVersion(ctx, expander.Version, expander.Type)
```
Comment (barney-s) [inline experiments/compositions/composition/internal/controller/expander_reconciler.go:184]: please handle error here. Similar to line 194 below.

```
	if err != nil {
		reason := "FailedGettingExpanderVersion"
		if apierrors.IsNotFound(err) {
			reason = "MissingExpanderVersion"
		}
		logger.Error(err, "Failed to get the ExpanderVersionCRD for %s, err: %s. Currently setting the version to v0.0.1.alpha", expanderType, reason)
		// The CR should be created before the specified expander can be used.
		newStatus.AppendErrorCondition(expander.Name, err.Error(), "....")
                 return ...
	}
```

**Review by barney-s — COMMENTED** [review]
Few comments. Please handle error at call site.

File: experiments/compositions/composition/tests/data/TestSimpleCompositionAddFacadeField/input.yaml (hunk 1 above)
Comment (xiaoweim) [reply experiments/compositions/composition/tests/data/TestSimpleCompositionAddFacadeField/input.yaml:111]: Yes, so for out testcases, I removed some version specification and some I left it there.

File: experiments/compositions/composition/internal/controller/expander_reconciler.go (hunk 3 above)
Comment (xiaoweim) [reply experiments/compositions/composition/internal/controller/expander_reconciler.go:326]: Thanks for pointing it out!

File: experiments/compositions/composition/internal/controller/expander_reconciler.go [hunk 5]
```diff
@@ -178,13 +181,25 @@ func (r *ExpanderReconciler) Reconcile(ctx context.Context, req ctrl.Request) (c
 		// ------------------- EXPANSION SECTION -----------------------
 		logger = loggerCR.WithName(expander.Name).WithName("Expand")
 
+		expanderVersion, expanderRegistry, err := r.getExpanderVersion(ctx, expander.Version, expander.Type)
+		if err != nil {
+			reason := "FailedGettingExpanderVersionCR"
+			if apierrors.IsNotFound(err) {
+				reason = "MissingExpanderVersionCR"
+			}
+			logger.Error(err, "Failed to get the ExpanderVersionCRD for %s, err: %s. Currently setting the version to v0.0.1.alpha", expander.Type, reason)
```
Comment (barney-s) [inline experiments/compositions/composition/internal/controller/expander_reconciler.go:190]: ExpanderVersionCR to be consistent.

**Review by barney-s — COMMENTED** [review]
I would recommend adding a test for testing "MissingExpanderVersionCR"

**Pull request merged**